### PR TITLE
Ks create get request

### PR DESCRIPTION
**Pull request by kms617**
Added endpoint to view an event via GET request.

### CODE REVIEW

**Comment by edwardloveall** [comment]
Something I meant to comment on @brimatteng too, in your PR description, you should also list other things you did aside from the main task. Such as:
- Converted double quotes to single quotes, app-wide

**Comment by edwardloveall** [comment]
Also, when you rebase, do an interactive rebase to make sure you get rid of things like `work in progress`. Those should never make it into master

File: app/assets/stylesheets/base/_forms.scss [hunk 1]
```diff
@@ -18,7 +18,7 @@ label {
   margin-bottom: $base-spacing / 4;
 
   &.required:after {
-    content: "*";
+    content: '*';
```
Comment (edwardloveall) [inline app/assets/stylesheets/base/_forms.scss]: You might have been a little zealous on the quotes. I'd say when starting a project, just take it on a file-by-file basis and don't worry about converting the whole app. You're much more likely to run into bugs this way.

Comment (edwardloveall) [inline app/assets/stylesheets/base/_forms.scss]: But good effort!

File: bin/spring [hunk 2]
```diff
@@ -4,15 +4,15 @@
...
   if match = Bundler.default_lockfile.read.match(/^GEM$.*?^    (?:  )*spring \((.*?)\)$.*?^$/m)
-    ENV["GEM_PATH"] = ([Bundler.bundle_path.to_s] + Gem.path).join(File::PATH_SEPARATOR)
-    ENV["GEM_HOME"] = ""
+    ENV['GEM_PATH'] = ([Bundler.bundle_path.to_s] + Gem.path).join(File::PATH_SEPARATOR)
+    ENV['GEM_HOME'] = ''
     Gem.paths = ENV
 
-    gem "spring", match[1]
-    require "spring/binstub"
+    gem 'spring', match[1]
```
Comment (edwardloveall) [inline bin/spring]: For example, I almost never touch anything in `/bin` unless someone tells me to. Too high probability for failure and you get really annoying errors if something gets screwy.

File: db/migrate/20150107222703_create_events.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class CreateEvents < ActiveRecord::Migration
+  def change
+    create_table :events do |t|
+      t.timestamps null: false
```
Comment (edwardloveall) [inline db/migrate/20150107222703_create_events.rb:4]: `null: false` :+1:

File: app/assets/stylesheets/base/_forms.scss (hunk 1 above)
Comment (kms617) [inline app/assets/stylesheets/base/_forms.scss]: ok, just changed this back

File: Gemfile [hunk 4]
```diff
@@ -1,56 +1,60 @@
...
+  gem 'shoulda-matchers', require: false
+  gem 'timecop'
+  gem 'webmock'
 end
 
 group :staging, :production do
-  gem "rails_stdout_logging"
+  gem 'airbrake'
+  gem 'newrelic_rpm'
+  gem 'rails_stdout_logging'
```
Comment (dthyresson) [inline Gemfile]: FYI, `rails_12factor` includes `rails_stdout_logging` so can omit `rails_stdout_logging` if you want.

File: bin/spring (hunk 2 above)
Comment (kms617) [inline bin/spring]: changed it back, too

File: Gemfile [hunk 5]
```diff
@@ -1,56 +1,60 @@
...
+gem 'coffee-rails', '~> 4.1.0'
+gem 'delayed_job_active_record'
+gem 'email_validator'
+gem 'flutie'
+gem 'high_voltage'
+gem 'jbuilder'
+gem 'jquery-rails'
+gem 'neat', '~> 1.5.1'
+gem 'normalize-rails', '~> 3.0.0'
+gem 'oj'
```
Comment (dthyresson) [inline Gemfile:17]: You may need `oj_mimic` here as 

https://github.com/ohler55/oj#rails

```
In version Rails 4.1, multi_json has been removed, and this patch is unnecessary and will no longer work. Instead, use the oj_mimic_json gem along with oj in your Gemfile to have Oj mimic the JSON gem and be used in its place by ActiveSupport JSON handling:
```

File: Gemfile (hunk 4 above)
Comment (kms617) [inline Gemfile]: done!

File: spec/factories.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  sequence :token do
+    SecureRandom.hex(3)
+  end
+
+  factory :event do
+    lat { Faker::Address.latitude.to_f.round(5) }
+    lon { Faker::Address.longitude.to_f.round(5) }
+    sequence(:name) { |n| "Name #{n}" }
+    sequence(:started_at) { |n| n.hours.from_now }
+    owner factory: :user
```
Comment (dthyresson) [inline spec/factories.rb]: So, there's another cool way to do this using FactoryGirl aliases.

`factory :user, alias [:owner] do`

and then can just rely on

```
     sequence(:started_at) { |n| n.hours.from_now }
     owner 
```

More: http://www.rubydoc.info/gems/factory_girl/file/GETTING_STARTED.md#Aliases

File: spec/i18n_spec.rb [hunk 7]
```diff
@@ -1,18 +0,0 @@
-require 'spec_helper'
```
Comment (edwardloveall) [inline spec/i18n_spec.rb:1]: Should add this file and the one that runs the test back. It's very helpful

File: app/models/event.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+class Event < ActiveRecord::Base
+  validates_presence_of :lat
```
Comment (edwardloveall) [inline app/models/event.rb:4]: Forgot about this, but we do associations before validations: https://github.com/IntrepidPursuits/web-style-guides/blob/master/style.md#rails

File: app/models/user.rb [hunk 9]
```diff
@@ -0,0 +1,4 @@
+class User < ActiveRecord::Base
+  validates :device_token, uniqueness: true
```
Comment (edwardloveall) [inline app/models/user.rb]: Same here with associations before validations

**Comment by edwardloveall** [comment]
Good to merge. Make sure you do an interactive rebase to squash some of those weird commits. Come get me if you've never done that.

**Pull request closed without merging**